axum/src/extract/path/de.rs

```diff
@@ -94,7 +94,20 @@ impl<'de> Deserializer<'de> for PathDeserializer<'de> {
                 .got(self.url_params.len())
                 .expected(1));
         }
-        visitor.visit_borrowed_str(&self.url_params[0].1)
+        let key = &self.url_params[0].0;
+        let value = &self.url_params[0].1;
+        visitor
+            .visit_borrowed_str(value)
+            .map_err(|e: Self::Error| match e.kind {
```

Comment:
Nit, I'm not sure you need to specify the type at all, but if you want to do that, I'd rather use `ErrorKind` for consistency's sake since you're using that on the next lines.

Reply:
The compiler barfs at me without annotation:
```
error[E0282]: type annotations needed
  --> axum/src/extract/path/de.rs:99:52
   |
99 |         visitor.visit_borrowed_str(value).map_err(|e| match e.kind {
   |                                                    ^        - type must be known at this point
   |
help: consider giving this closure parameter an explicit type
   |
99 |         visitor.visit_borrowed_str(value).map_err(|e: /* Type */| match e.kind {
   |                                                     ++++++++++++

For more information about this error, try `rustc --explain E0282`.
error: could not compile `axum` (lib) due to 1 previous error
```

I will change it to `ErrorKind`.